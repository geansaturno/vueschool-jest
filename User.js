export default class User {
  constructor ({ firstname, lastname }) {
    this.firstname = firstname
    this.lastname = lastname
  }

  get name () {
    return `${this.firstname} ${this.lastname}`
  }
}
