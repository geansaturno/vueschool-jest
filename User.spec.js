import User from './User.js'

describe('User', () => {
  it('Should show the name correct', () => {
    const user = new User({ firstname: 'Gean', lastname: 'Gonçalves' })

    expect(user.name).toBe('Gean Gonçalves')
    expect(1).toEqual(expect.any(Number))
    expect(user).toMatchSnapshot()
  })
})
