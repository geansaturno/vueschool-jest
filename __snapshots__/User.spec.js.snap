// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Should show the name correct 1`] = `
User {
  "firstname": "Gean",
  "lastname": "Gon√ßalves",
}
`;
